konig-core/src/main/java/io/konig/shacl/PropertyConstraint.java

```diff
@@ -84,6 +84,7 @@
 	
 	private Term term;
 	private URI termStatus;
+	private boolean required = true;
```

Comment:
I'm not comfortable adding the "required" field to PropertyConstraint.   This value is derived from minCount.  See my comment below on the implementation of the isRequired method.